cmd/minishift/cmd/hostfolder/add.go

```diff
@@ -56,6 +69,11 @@ var hostfolderAddCmd = &cobra.Command{
 func init() {
 	HostfolderCmd.AddCommand(hostfolderAddCmd)
 	hostfolderAddCmd.Flags().BoolVarP(&instanceOnly, "instance-only", "", false, "Defines the host folder only for the current OpenShift cluster.")
+	hostfolderAddCmd.Flags().StringVarP(&unc, "unc", "", "", "The UNC remote path of the share")
+	hostfolderAddCmd.Flags().StringVarP(&username, "username", "", "", "Username of the shared CIFS server")
+	hostfolderAddCmd.Flags().StringVarP(&password, "password", "", "", "Password of the shared CIFS server")
+	hostfolderAddCmd.Flags().StringVarP(&mountPoint, "mountpoint", "", "", "Target mount point inside the VM")
+	hostfolderAddCmd.Flags().StringVarP(&domain, "domain", "", "", "Domain")
```

Comment:
@hferentschik We can also add a `--type` flag where we spacify if its CIFS or sshfs, but then again we'll have different parameteres for them individually, i think username and password will be required for sshfs also.. right?

Reply:
Different parameters are needed. They can be created from metadata you provide. All options are specific to a type. Hostdir has no meaning to Windows users, as this can also be a remotedir... UNC is their terminology.

Reply:
> i think username and password will be required for sshfs also

I hope not

Reply:
> All options are specific to a type.

+1

 > Hostdir has no meaning to Windows users, as this can also be a remotedir... UNC is their terminology.

This is a tricky one. Really hostfolders is about getting a folder on your host mounted into your VM. In this context I see this together as the mountpoint as the two unifying options. Admittedly UNC is the Windows term used for shares, but I was hoping we can unify this in some form maybe with different names. In the end there is always a source and a destination of the mount. There there is the type of course. A part of this, we are probably dealing with type specific options.

Reply:
> I hope not

Some form of authentication is needed. If not password, a possible location to the certificate for login would be needed.

Reply:
> UNC

Actually, the term UNC stands for Uniform Naming Convention and is part of the CIFS standard. It seems some operating systems (macOS in this case) tends to avoid mentioning this. However, since we support CIFS, I'd rather stick to the official terminology.